url.bs

```diff
@@ -1400,6 +1400,9 @@ string <var>input</var>, optionally with a <a>base URL</a> <var>base</var>, opti
 
        <li><p>Set <var>url</var>'s <a for=url>scheme</a> to <var>buffer</var>.
 
+       <li><p>If <var>url</var>'s <a for=url>port</a> is <var>url</var>'s <a for=url>scheme</a>'s
+       <a>default port</a>, then set <var>url</var>'s <a for=url>port</a> to null.
```

Comment:
This step is required for the protocol setter only, so it's better to place it after "4. If state override is given"

2. Set url ’s scheme to buffer
3. Set buffer to the empty string
4. If state override is given, then:
   1. **If url ’s port is url’s scheme’s default port , then set url’s port to null**.
   2. Return.